scripts/update-tezos.sh

```diff
@@ -0,0 +1,48 @@
+#! /usr/bin/env bash
+# SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+# This script fetches the latest tag from the https://gitlab.com/tezos/tezos/ repository,
+# compares it with the version presented in the nix/nix/sources.json, and performs an
+# update if the versions are different
+
+set -e
+
+git config user.name "Serokell CI bot" # necessary for pushing
+git config user.email "hi@serokell.io"
+git fetch --all
+
+# Get latest tag from tezos/tezos
+git clone https://gitlab.com/tezos/tezos.git upstream-repo
+cd upstream-repo
+latest_upstream_tag_hash="$(git rev-list --tags --max-count=1)"
+latest_upstream_tag="$(git describe --tags "$latest_upstream_tag_hash")"
+source scripts/version.sh
+cd ..
+rm -rf upstream-repo
+
+branch_name="auto/$latest_upstream_tag-release"
+
+our_tezos_tag="$(jq -r '.tezos.ref' nix/nix/sources.json | cut -d'/' -f3)"
+
+if [[ "$latest_upstream_tag" > "$our_tezos_tag" ]]; then
```

Comment:
:thinking: AFAIR there sometimes were periods, e.g. around v10 in August, when first a v9.7 came out, then v10.0-rc1, then v9.8 ("decreasing" the tag). AFAIU this comparison can only handle monotonically increasing tags, not this situation? Maybe instead of `>` we can get away with just `!=`?